informational/req-res-reliability.md

```diff
@@ -0,0 +1,89 @@
+
+### Definitions
+- Service node - provides services to other nodes such as relaying messages send by LightPush to the network or broadcasts messages from the network through Filter, usually serves responses;
+- Light node - connects to and uses one or more service nodes via LightPush and/or Filter protocols, usually sends requests;
+
+## Motivation
+
+Specifications of the mentioned protocols do not define some of the real world use cases that are often observed in unreliable network environment from the perspective of light nodes that are consumers of LightPush and/or Filter protocols.
+Such use cases can be: recovery from offline state, decrease rate of missed messages, increase probability of messages being broadcasted within the network.
+
+## Suggestions
+
+### Node health
+
+Node health is a metric meant to determine how reliable a light node is.
+We consider this reliability to be dependant on amount of simultaneous connections to responsive service nodes.
+Unfortunately the more connections light node establishes - the more bandwidth is consumed.
+To address this we suggest following metrics:
+- unhealthy - no connections to service nodes are available regardless of protocol;
+- minimally healthy:
+  - Filter has one service node connection;
+  - LightPush protocol has one service node connection;
+- sufficiently healthy:
+  - Filter has at least 2 connections available to service nodes;
+  - LightPush has at least 2 connections available to service nodes;
+
+### Peers and connection management
+
+#### Pool of reliable service nodes
+Light node should maintain a pool of reliable service nodes for each protocol.
+In case service node fails to serve protocol request from a light node 3 times - light node should drop connection to it and a new service node should be connected and added to the pool instead.
+
+#### Seection of discovered service nodes
+During discovery light node should filter out service nodes based on preferences before establishing connection.
+These preferences might include:
+- [Libp2p multiadresses](https://github.com/libp2p/specs/blob/master/addressing/README.md) of a service node;
+- Waku or libp2p protocols that a service node implements;
+- Wakus shards that a service node is part of;
+
+More details about discovery can be found at [WAKU2 Discovery domain](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/10/waku2.md#discovery-domain) or [RELAY-SHARDING Discovery](https://github.com/waku-org/specs/blob/master/standards/core/relay-sharding.md#discovery).
+
+Examples of filtering:
+- When [Circuit V2](https://github.com/libp2p/specs/blob/master/relay/circuit-v2.md) multiaddresses discovered by a light node - it might avoid connecting such service nodes and wait for service nodes that can be connected directly;
+- When light node discovers service nodes that implement needed Waku protocols - it should prioritize those that implement most recent version of protocol;
```

Comment:
At this point in time, we should aim for single version protocol support in light clients IMO.

We can look into multiple version support as the network grows and we have less control on it.

Reply:
I think it is still worth mentioning as it might be not only version but needed protocol depending on the strategies used: Store + LightPush or Filter + LightPush or all